src/bokeh/embed/server.py

```diff
@@ -117,11 +120,15 @@ def server_document(url: str = "default", relative_urls: bool = False, resources
 
     headers = headers or {}
 
+    if headers and with_credentials:
+        warnings.warn()
```

Comment:
Unfortunately, these two currently don't work together and likely never will. The issue is that if the browser sends credentials, the `Access-Control-Allow-Headers *` header is no longer valid. In this case we have to explicitly enumerate all non-standard headers that will be send. However, since there is no way to pass the headers we receive here to the `AutoloadJsHandler`, we cannot do that.

LMK if I should turn this into an error instead. I'll also add something to the docstring above, when we have reached consensus here.

Reply:
Let's make it an error.

Reply:
+1 on making it an error.